### PR TITLE
Adjustments for shorten-64-to-32 warnings

### PR DESCRIPTION
**Pull request by jtroehr**
Ok to proceed with issue #710 like that?

### CODE REVIEW

**Comment by h-2** [comment]
Some thoughts: some times you have replaced int with long, but on Windows this will still be 32bit. If you are dealing with function return values, please check what they return, e.g. `strlen()` returns a value of type `std::size_t` so please use that instead of hardcoding to `long`. This is also true for operations on pointers which often resolve to size_t in the STL.
More comments above.

File: include/seqan/basic/debug_test_system.h [hunk 1]
```diff
@@ -754,7 +754,7 @@ void beginTestSuite(const char * testSuiteName, const char * argv0)
     const char * ptr = std::min(strchr(argv0, '\\'), strchr(argv0, '/'));     // On Windows, we can have both \ and /.
     for (; ptr != 0; ptr = std::min(strchr(ptr + 1, '\\'), strchr(ptr + 1, '/')))
         end = ptr;
-    int rpos = end - argv0;
+    long rpos = end - argv0;
     if (rpos <= 0)
```
Comment (h-2) [inline include/seqan/basic/debug_test_system.h]: use `std::size_t``

File: include/seqan/basic/debug_test_system.h [hunk 2]
```diff
@@ -1658,8 +1658,8 @@ verifyCheckPoints(const char * file)
 
 
 
-    int len = strlen(StaticData::pathToRoot()) +
-              strlen("/") + strlen(file) + 1;
+    unsigned long len = strlen(StaticData::pathToRoot()) +
+                        strlen("/") + strlen(file) + 1;
     char * absolutePath = new char[len];
```
Comment (h-2) [inline include/seqan/basic/debug_test_system.h]: as well in the above to cases

File: include/seqan/file/file_page.h [hunk 3]
```diff
@@ -888,7 +888,7 @@ struct MMap;
     }
 
     template <typename TValue, typename TFile> inline
-    unsigned readBucket(PageBucket<TValue> &b, int pageNo, size_t pageSize, size_t dataSize, TFile &file)
+    unsigned long readBucket(PageBucket<TValue> &b, int pageNo, size_t pageSize, size_t dataSize, TFile &file)
     {
```
Comment (h-2) [inline include/seqan/file/file_page.h]: size_t as well

File: include/seqan/file/file_page.h [hunk 4]
```diff
@@ -1305,7 +1305,7 @@ firstToEnd(PageChain<TPageFrame> &pageChain)
         // Function is a functor which is called with a PageFrame object,
         // that is dirty or not READY (in an IO transfer)
         template <class Function>
-        inline int mru(Function Func_, unsigned maxLevel = PRIORITY_LEVELS - 1)
+        inline unsigned long mru(Function Func_, unsigned maxLevel = PRIORITY_LEVELS - 1)
         {
```
Comment (h-2) [inline include/seqan/file/file_page.h]: why did you add unsigned here? The function actually can return -1 !! Also what types does PageLRUList acutally cover?

File: include/seqan/file/file_page.h [hunk 5]
```diff
@@ -1515,7 +1515,7 @@ firstToEnd(PageChain<TPageFrame> &pageChain)
     }
 
     template <typename TValue, typename TSize, typename T, class Function>
-    inline unsigned equiDistantAlignedDistribution(
+    inline unsigned long equiDistantAlignedDistribution(
         Buffer<TValue> &_clusterBuffer, size_t aligning, size_t _bufferSize, T const &me,
```
Comment (h-2) [inline include/seqan/file/file_page.h]: size_t

File: include/seqan/graph_align/graph_impl_align.h [hunk 6]
```diff
@@ -1656,7 +1656,7 @@ convertAlignment(Graph<Alignment<TStringSet, TCargo, TSpec> > const& g,
     TSize col = 0;
     typename TPosToVertexMap::const_iterator it = g.data_pvMap.begin();
     unsigned int compIndex = 0;
-    unsigned int compIndexLen = length(order);
+    unsigned long compIndexLen = length(order);
     TIdType currentSeq = it->first.first;
```
Comment (h-2) [inline include/seqan/graph_align/graph_impl_align.h]: Since order is of type `String<unsigned int>` the return value of length is of type `typename Position<String<unsigned int> >::Type`.

File: include/seqan/random/random_mt19937.h [hunk 7]
```diff
@@ -84,7 +84,7 @@ class Rng<MersenneTwister>
     { SEQAN_CHECKPOINT; }
 
     inline
-    unsigned
+    unsigned long
     operator()()
```
Comment (h-2) [inline include/seqan/random/random_mt19937.h]: You don't need to change anything in the random module because it will be removed entirely for 2.1.0.

**Comment by h-2** [comment]
Sorry for the many comments, but I think these are the standard cases. When dealing with standard types it is usually size_t and with our types it is usually `typename Position< T>::Type`.

File: include/seqan/graph_align/graph_impl_align.h [hunk 8]
```diff
@@ -1635,6 +1635,7 @@ convertAlignment(Graph<Alignment<TStringSet, TCargo, TSpec> > const& g,
     typedef typename Size<TGraph>::Type TSize;
     typedef typename Id<TGraph>::Type TIdType;
     typedef typename TGraph::TPosToVertexMap_ TPosToVertexMap;
+	typedef typename Position<String<unsigned int> >::Type TOrderLength;
```
Comment (jtroehr) [inline include/seqan/graph_align/graph_impl_align.h]: Will use spaces...

**Comment by h-2** [comment]
@jtroehr was ist hier der Status? Danke!

**Comment by jtroehr** [comment]
@h-2 noch nicht weit, plane am Montag zur session zu kommen.

**Comment by h-2** [comment]
@jtroehr we want to release tomorrow afternoon, what is the status here? Thanks.

**Comment by jtroehr** [comment]
@h-2 Changes are committed. PR could be reviewed and merged. But the issue is by far not done. I counted the number of remaining warnings in seqan/include and there are still about 25,000 ;) after substracting warnings for the random module. Which other modules will be removed?

**Comment by jtroehr** [comment]
Won't be around today as we are close to manuscript submission...

**Comment by h-2** [comment]
@jtroehr @marehr can you coordinate on this? thanks.

**Comment by xenigmax** [comment]
This can't be done within 2 days so I moved it to this to 2.1.2

**Comment by marehr** [comment]
Superseded by #1734
